### PR TITLE
dev/core#1511 Expose option to show event location on configuration tab

### PR DESCRIPTION
**Pull request by aydun**
Overview
----------------------------------------
Add a checkbox on the Event Location configuration tab to control whether the location is displayed to users.

The functionality already exists and the field can be configured by API.  This enables configuration via the GUI.

Before
----------------------------------------
The option can only be configured by API.

After
----------------------------------------
Option can also be configured via GUI.

image

Technical Details
----------------------------------------
The logic for the use of the option already exists.  This change just exposes it via GUI.

Comments
----------------------------------------
See https://lab.civicrm.org/dev/core/issues/1511

### CODE REVIEW

**Comment by yashodha** [comment]
@adyun looks good, tested this and it works fine.

**Comment by mattwire** [comment]
@aydun The only issue I have here is reading "Show Location?" I have no idea *where* it is actually shown.  If you can clarify via text/description/help text then I'm happy to merge this.

**Comment by yashodha** [comment]
@mattwire @aydun Would it make sense it to provide the event info link (where the location will be showed or not) so that it is more clear. Something like

Show location [here](https://dmaster.demo.civicrm.org/civicrm/event/info?reset=1&id=3)

**Comment by aydun** [comment]
It affects the location in a few places - I'll add a help note.

**Comment by aydun** [comment]
Help text added - please re-review.

File: templates/CRM/Event/Form/ManageEvent/Location.hlp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+{htxt id="id-is_show_location-title"}
+  {ts}Show Location{/ts}
+{/htxt}
+{htxt id="id-is_show_location"}
+  {ts}If this option is selected, the location will be shown in various places including:
+    <ul>
+      <li>the Event Information page</li>
+      <li>event registration confirmation emails</li>
+      <li>the HTML, RSS and ICALENDAR feeds of event information</li>
+    </ul>
```
Comment (yashodha) [inline templates/CRM/Event/Form/ManageEvent/Location.hlp]: @aydun 
Can we remove the in first and last and capitalise all or none

- Event Information page
- Event registration confirmation emails
- HTML, RSS and ICALENDAR feeds of event information

**Comment by mattwire** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
(*CiviCRM Review Template DEL-1.2*)


* General standards
    * Explain ([`r-explain`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-explain))
        * __PASS__ : The goal/problem/solution have been adequately explained with a link (JIRA, Github, Gitlab, StackExchange).
    * User impact ([`r-user`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-user))
        * __PASS__: The change would be intuitive or unnoticeable for a majority of users who work with this feature.
    * Documentation ([`r-doc`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-doc))
        * __PASS__: The changes do not require documentation. There is help text and we checked the  docs & didn't see an outdated screenshot
    * Run it ([`r-run`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-run))
        * __PASS__: We tested this & setting this variable altered how it shows up on the info screen. We weren't able to figure  out how to test  that  it   shows on  the  email - on the test  site

    * Technical impact ([`r-tech`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-tech))
        * __PASS__: The change preserves compatibility with existing callers/code/downstream.
    * Code quality ([`r-code`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-code))
       * __PASS__: 
    * Maintainability ([`r-maint`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-maint))

        * __PASS__: The change sufficiently improves test coverage, or the change is trivial enough that it does not require tests because it is only the tpl  layer
    * Test results ([`r-test`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-test))

        * __PASS__: The test results have failures, but these have been individually inspected and found to be irrelevant.

**Comment by eileenmcnaughton** [comment]
test this pleasee

**Comment by seamuslee001** [comment]
Test failures unrelated

**Comment by MikeyMJCO** [comment]
Will this drop in 5.23 or 5.24 -> I suspect 5.24 but am not always entirely sure...

**Comment by eileenmcnaughton** [comment]
5.23 - 5.22 is the current rc so itt will  hit the  next one

**Pull request merged**